_posts/2024-06-22-custom-copr-builders.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: "Need something extra? Employ your own Copr builders!"
+date: 2024-06-22 00:00:00 +0000
+lang: en
+---
+
+In Copr, we deploy the [Resalloc server][resalloc] to manage our farm of
+builders.  The _Copr_ build system (specifically the _copr-backend_ package)
+acts as the "client," creating tickets with _Resalloc_ and requesting specific
+kinds of worker virtual machines ("resources") for particular build tasks.
+To do this, _Copr_ uses a known set of per-ticket tags to describe particular
+needs (e.g., `--tag arch_x86_64`, `--tag powerful`, and so on).
+
+_Resalloc's_ task is to match the _client's_ ticket tags with pre-allocated VMs
+(or allocate new VMs if needed).
```

Comment:
I'd probably suggest dumbing down the first two paragraphs. There is IMHO too much detail about tags, copr-backend as a client, and so on. I'd rather mention that our farm consists of builders in multiple clouds and multiple architectures. And what are the current limitations (for payoff when mentioning the feature in later paragraphs) like having to wait in the queue and pre-configured VM specs.

Reply:
Makes sense, going to drop useless theory/details!
Having to wait in queue? There's no competition with other users in this case :shrug: so users should not wait..

Reply:
> Having to wait in queue? There's no competition with other users in this case 🤷 so users should not wait..

I mean having to wait in queue as a normal user, and therefore motivation for somebody to add custom builders (having no competition in the queue).

Reply:
Thanks. Updated, marked as resolved.